### PR TITLE
Remove the USE_MIOPEN build option as MIOpen is always used when built with ROCm.

### LINKED ISSUE #22200: USE_MIOPEN is not a proper CMake variable
This is mentioned in `setup.py` as configurable, but per `tools/setup_helpers/miopen.py` the actual flag is `NO_MIOPEN`. Also, if you set `USE_MIOPEN=0` when compiling, `setup.py` fails when it tries to build the Python extension. This is caused by CMake not having type information for the USE_MIOPEN variable, because it is not defined anywhere.

### PR DESCRIPTION
**Pull request by xuhdev**
Close #22200

### CODE REVIEW

File: torch/CMakeLists.txt [hunk 1]
```diff
@@ -194,12 +194,6 @@ if (USE_CUDNN)
     list(INSERT 0 TORCH_PYTHON_INCLUDE_DIRECTORIES ${CUDNN_INCLUDE_DIR})
 endif()
 
-if (USE_MIOPEN)
```
Comment (bddppq) [inline torch/CMakeLists.txt]: these compile options are still needed, replace `USE_MIOPEN` with `USE_ROCM`

Comment (xuhdev) [reply torch/CMakeLists.txt]: As discussed in #22227, things in this if-clause have never been invoked. In fact, experiment in #22345 shows that setting `USE_MIOPEN=USE_ROCM` fails ROCm builds. I would see this block as something that was forgotten to be removed.

Comment (xuhdev) [reply torch/CMakeLists.txt]: Hmm, the CI showed the same error. I've made changes according to your suggestion. Let's wait for the CI...

Comment (bddppq) [reply torch/CMakeLists.txt]: Oh I know what's going on - it's `CAFFE2_USE_CUDNN` gets hipified to `CAFFE2_USE_MIOPEN`

**Review by bddppq — DISMISSED** [review]

**Comment by bddppq** [comment]
Benchmark runs:
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/32006/
https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-devtoolset7-rocmrpm-centos7.5-trigger-test/30889/

File: torch/CMakeLists.txt (hunk 1 above)
Comment (xuhdev) [reply torch/CMakeLists.txt]: I modified #22345 and confirmed that this if block is never executed for ROCm build. ("MIOpen flags are enabled!!!", which indicates the execution of this if block, does not show up in the CI output.)

Comment (xuhdev) [reply torch/CMakeLists.txt]: I also removed a previous deletion of `CAFFE2_USE_MIOPEN`, which broke the build, just as you said.

**Review by bddppq — APPROVED** [review]
LG

**Comment by pietern** [comment]
Thanks @xuhdev!

**Pull request closed without merging**